.github/workflows/pull-request.yml

```diff
@@ -47,3 +47,30 @@ jobs:
         same_commit: True
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+  CodeQL:
```

Comment:
Add a `needs` line so that we can determine the order of execution.

1. `Code-Formatter` should need `Linter`
2. `CodeQL` should need `Code-Formatter`
3. `Count-lines-of-code` should need `CodeQL`